### DISCUSSION TITLE
Support for NRF52xxx in the MP3_Decoder

### DISCUSSION THREAD

**Discussion by aovestdipaperino** [schreibfaul1/ESP32-audioI2S#441] [Ideas]
Hello, I was able to add support for NRF52xxx to the decoder which is simply:

- removing the log_e and log_i statements (need to find a replacement)
- use the standard ``malloc``

I've put mine [here](https://github.com/aovestdipaperino/NRF_I2SPlayer), it would be interesting if the mp3_decoder.* can somewhat be kept in sync (maybe a different repo?)

**Comment by schreibfaul1** [comment]
The NRF52xx are Bluetooth modules, right? The BT integrated in the ESP32 only works properly with a sample rate of 44100Hz. Can you also use other sample rates, e.g. 48000 or 22050 with NRF52?

**Comment by aovestdipaperino** [comment]
Yes, different rates are supported.
See table [here](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.nrf52832.ps.v1.1%2Fi2s.html)
It's not exhaustive as I've seen 22.05KHz mentioned as well in some code.

**Comment by schreibfaul1** [comment]
Thank you for your feedback. Then I'll get a NRF51822, there are 2 on Amazon for 10€, that's affordable, and will experiment with it. I have your sample code as a template :-))

**Reply by aovestdipaperino** [reply]
The Seeed Xiao ones are very good.
